lib/LedgerSMB/Scripts/setup.pm

```diff
@@ -779,72 +782,65 @@ sub _failed_check {
     my $hiddens = {
        check => $check->name,
 verify_check => md5_hex($check->test_query),
-    database => $request->{database}
+    database => $request->{database},
     };
+    my @skip_keys = grep /^skip_/, keys %$request;
+    $hiddens->{@skip_keys} = $request->{@skip_keys};
 
     my $rows = [];
     while (my $row = $sth->fetchrow_hashref('NAME_lc')) {
-      my $id = $row->{$check->{id_column}};
+      my $count = 1+scalar(@$rows);
+      #TODO: A '' will fail with invalid input syntax in fix_tests - YL
+      my $id = join(',',map { $row->{$_} // '' } @{$check->id_columns});
```

Comment:
What happens when the value in the primary key is a character field containing a comma? (Or do we only have numeric p-keys so far?)

Reply:
ID of columns are SQL identifiers

Reply:
id_columns contains the columns id, not data. So, unless we want to protect ourselves in case of someone starts to use quoted ids, which we don't elsewhere, joining the id array with commas is totally acceptable.

Reply:
Sure. But `map { $row->{$_} // '' }` maps the column names into the data for the current record. Which means you can get any content, including commas.